### PR TITLE
Add UI for soul transfer

### PR DESCRIPTION
**Pull request by brian-man**
This change adds a new Transfer tab to the existing Burn/Upgrade set in the Plaza.

Contract support for transferring souls was in the initial mainnet release of the Character Burning feature, but there was no direct UI support previously.

### Screenshots

1. Opening `Plaza` page
image

2. After clicking `Burn/Upgrade/Transfer`
image

3. After clicking `Transfer`
image

4. After clicking `Use max` and entering a partial address
image

5. After completing the address
image

6. After clicking `Transfer souls`
image

### Testing

Basic functionality validated as follows:
1. The `invalid address` message is working correctly.
2. The `Use max` works as expected.
3. Confirmation dialog looks as expected, including the soul amount shown.
4. The contract method is called.
5. The sender amount is decremented the correct amount.
6. The receiver amount is incremented the correct amount.

### CODE REVIEW

**Review by kcper — APPROVED** [review]
Looks good.

File: frontend/src/views/Plaza.vue [hunk 1]
```diff
@@ -31,23 +31,30 @@
...
+              @click="isBurning = false; isUpgrading = true; isTransferring = false; clearAllBurn();"
             >
               <span>{{$t('plaza.upgrade')}}</span>
             </button>
+            <button
+              class="switch"
+              :class="{ switch_active: isTransferring }"
+              @click="isBurning = false; isUpgrading = false; isTransferring = true; clearAllBurn();"
+            >
+              <span>Transfer</span>
```
Comment (Chriserus) [inline frontend/src/views/Plaza.vue:51]: Missing translation key

File: frontend/src/views/Plaza.vue [hunk 2]
```diff
@@ -66,6 +73,15 @@
...
             ({{soulAmount}} {{$t('plaza.soul')}})
           </b-button>
+          <b-button
+            v-if="isTransferring"
+            variant="primary"
+            class="ml-3"
+            @click="showTransferConfirmation"
+            v-tooltip="$t('plaza.transferSelected')"
+            :disabled="!isValidWeb3Address(receiverAddress)">
+            Transfer souls
```
Comment (Chriserus) [inline frontend/src/views/Plaza.vue:83]: Same here

File: frontend/src/views/Plaza.vue [hunk 3]
```diff
@@ -149,6 +167,12 @@
                   <h2 :class="powerLimitExceeded ? 'text-danger' : ''">{{$t('CharacterDisplay.power')}} +{{burnPower}}/{{remainingPowerLimit}}</h2>
                 </div>
               </div>
+              <div v-if="isTransferring">
+                <b-form-input placeholder="Receiver address" v-model="receiverAddress"/>
+                <div class="transferResultContainer">
+                  <span class="resultMsg text-center"> {{(!isValidWeb3Address(receiverAddress) && receiverAddress !== '') ? 'Invalid address' : ''}} </span>
```
Comment (Chriserus) [inline frontend/src/views/Plaza.vue:173]: isValidWeb3Address might also check for empty string, please check, if not, leave it as-is. Missing translation "Invalid address"

Comment (brian-man) [reply frontend/src/views/Plaza.vue:173]: I wouldn't expect it to consider empty as valid.  The empty check was modeled on other code that did that check.

File: frontend/src/views/Plaza.vue (hunk 1 above)
Comment (brian-man) [reply frontend/src/views/Plaza.vue]: Fixed

File: frontend/src/views/Plaza.vue (hunk 2 above)
Comment (brian-man) [reply frontend/src/views/Plaza.vue]: Fixed

**Review by Chriserus — APPROVED** [review]

**Pull request merged**